Sources/Bookmarks/FaviconsFetcher/BookmarksFaviconsFetcherStateStore.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  BookmarksFaviconsFetcherStateStore.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public class BookmarksFaviconsFetcherStateStore: BookmarksFaviconsFetcherStateStoring {
+
+    let dataDirectoryURL: URL
+    let missingIDsFileURL: URL
+
+    public init(applicationSupportURL: URL) throws {
+        dataDirectoryURL = applicationSupportURL.appendingPathComponent("FaviconsFetcher")
+        missingIDsFileURL = dataDirectoryURL.appendingPathComponent("missingIDs")
+
+        try initStorage()
+    }
+
+    private func initStorage() throws {
+        if !FileManager.default.fileExists(atPath: dataDirectoryURL.path) {
+            try FileManager.default.createDirectory(at: dataDirectoryURL, withIntermediateDirectories: true)
```

Comment:
We don't _need_ it because this class is not initialized in BSK, but in client apps. The sole reason for having `BookmarksFaviconsFetcherError` is to be able to pass various errors to client apps via `EventMapping< BookmarksFaviconsFetcherError>`. For this function we don't need EventMapping because it's initialized directly in client apps that are expected to handle the error with a pixel as appropriate.